src/Core/Tokens/Token.cs

```diff
@@ -26,11 +27,28 @@ public Token RemovePrefix(string expectedPrefix)
         return new Token(_token[expectedPrefix.Length..]);
     }
 
-    public Token ProtectWith(IDataProtector dataProtector) =>
-        new(dataProtector.Protect(ToString()));
 
-    public Token UnprotectWith(IDataProtector dataProtector) =>
-        new(dataProtector.Unprotect(ToString()));
+    public Token ProtectWith<T>(IDataProtector dataProtector, ILogger<T> logger)
+    {
+        logger.LogDebug("Protecting token: {0}", this);
+        return new(dataProtector.Protect(ToString()));
+    }
+
+    public Token UnprotectWith<T>(IDataProtector dataProtector, ILogger<T> logger)
+    {
+        var unprotected = "";
+        try
+        {
+            unprotected = dataProtector.Unprotect(ToString());
+        }
+        catch (Exception e)
+        {
+            logger.LogInformation(e, "Failed to unprotect token: {0}", this);
```

Comment:
What's the output of logging `this` to a string in this way? Do you just get a key:value representation of the `Token` instance properties?

Reply:
Good catch!  @MGibson1 , I think you can change `{0}` to {@0}` to tell Serilog to deconstruct `this`. See https://bitwarden.atlassian.net/browse/DEVOPS-949?focusedCommentId=25722 for an example.

Reply:
@MGibson1 I adjusted this message and all the others to include the `@` operator for destructuring of the objects.

Reply:
It depends on the logger implementation, but the preferred way of dealing with these loggers is to have format-able strings rather than format them yourself. https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-message-template

You're also supposed to use names, not numbers. I'll make that change